streaming/streaming.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package streaming
```

Comment:
Considering this has to do with the data migration. Might make sense to keep this under sources/common instead

Reply:
For profile specific information, we can keep it in profiles/source_profile or create a new profile/streaming_profile

Reply:
sources/common brings a cyclic dependency hence we cannot put it there. We could create a new package in sources , but thinking this is not really a source specific module but more of a general one, I feel it should lie outside on a top level folder.